### PR TITLE
Dump conv args into file

### PR DESCRIPTION
**Pull request by pyjhzwh**
Summary: 
In the FX graph extracted from python key tracing, when seeing calls to aten.convolution, dump args including shapes (for input/bias/weight) and different options such as padding/stride/kernel size/dilation/etc to a json file.

Usage:
```
with torchdynamo.optimize(conv_args_analysis):
   func_to_analysis()
```
Test:
`pytest tests/test_optimizations.py::TestOptimizations::test_log_conv_args`

### CODE REVIEW

File: torchdynamo/optimizations/log_args.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    def run_node(self, n: torch.fx.Node):
+        result = super().run_node(n)
+
+        if n.op == "call_function":
+            if n.target == aten.convolution:
+                args, kwargs = self.fetch_args_kwargs_from_env(n)
+                assert len(args) == 9, "aten.convolution should have 9 args"
+                conv_args = {}
+                conv_args["input"] = args[0].shape
```
Comment (jansel) [inline torchdynamo/optimizations/log_args.py:42]: Perhaps also record `args[n].stride()`, so we can tell the difference between channels first vs last.

For completeness `args[n].dtype` might also be useful.

**Review by jansel — COMMENTED** [review]
Overall looks good.  Are you running this with torchbench.py models?

**Comment by jansel** [comment]
For the test failures (missing functorch), you can add a `unittest.skipIf` functorch is missing.

**Comment by pyjhzwh** [comment]
> Overall looks good. Are you running this with torchbench.py models?

Added option `--log-conv-args` in torchbench.py

**Review by jansel — APPROVED** [review]

**Pull request merged**